deployment/terraform/create.go

```diff
@@ -160,22 +175,75 @@ func (t *Terraform) Create() error {
 
 			// Starting mattermost.
 			mlog.Info("Starting mattermost", mlog.String("host", ip))
-			cmd := fmt.Sprintf("sudo service mattermost start")
+			cmd := "sudo service mattermost start"
 			if err := sshc.RunCommand(cmd); err != nil {
 				mlog.Error("error running ssh command", mlog.String("cmd", cmd), mlog.Err(err))
 				return
 			}
 		}()
 	}
+}
 
-	// TODO: display the entire cluster info from terraformOutput later
-	// when we have cluster support.
-	mlog.Info("Deployment complete.")
+func (t *Terraform) setupProxyServers(output *terraformOutput, extAgent *ssh.ExtAgent) {
+	for _, ip := range output.ProxyIP.Value {
+		sshc, err := extAgent.NewClient(ip)
+		if err != nil {
+			mlog.Error("error in getting ssh connection", mlog.String("ip", ip), mlog.Err(err))
+			continue
+		}
+		func() {
+			defer func() {
+				err := sshc.Close()
+				if err != nil {
+					mlog.Error("error closing ssh connection", mlog.Err(err))
+				}
+			}()
 
-	return nil
+			// Upload service file
+			mlog.Info("Uploading nginx config", mlog.String("host", ip))
+
+			backends := ""
+			for _, addr := range output.Instances.Value {
+				backends += "server " + addr.PrivateIP + ":8065;\n"
+			}
+
+			files := []struct {
+				path    string
+				content string
+			}{
+				{content: strings.TrimSpace(fmt.Sprintf(nginxConfig, backends)), path: "/etc/nginx/sites-available/mattermost"},
+				{content: strings.TrimSpace(sysctlConfig), path: "/etc/sysctl.conf"},
+				{content: strings.TrimSpace(limitsConfig), path: "/etc/security/limits.conf"},
+			}
+			for _, fileInfo := range files {
+				rdr := strings.NewReader(fileInfo.content)
+				if err := sshc.Upload(rdr, fileInfo.path, true); err != nil {
+					mlog.Error("error uploading file", mlog.Err(err), mlog.String("file", fileInfo.path))
+					return
+				}
+			}
+
+			for _, cmd := range []string{
+				"sudo ln -fs /etc/nginx/sites-available/mattermost /etc/nginx/sites-enabled/mattermost",
+				"sudo rm -f /etc/nginx/sites-enabled/default",
+				"sudo grep -q -F 'worker_rlimit_nofile' /etc/nginx/nginx.conf || echo 'worker_rlimit_nofile 65536;' | sudo tee -a /etc/nginx/nginx.conf",
+				"sudo sed -i 's/worker_connections.*/worker_connections 200000;/g' /etc/nginx/nginx.conf",
+				"sudo systemctl daemon-reload",
```

Comment:
The daemon-reload and enable parts can be done in the remote-exec provisioning itself.